lib/charms/mutual_tls_interface/v0/mutual_tls.py

```diff
@@ -0,0 +1,346 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Library for the mutual-tls relation.
+
+This library contains the Requires and Provides classes for handling the mutual-tls interface.
+
+## Getting Started
+From a charm directory, fetch the library using `charmcraft`:
+
+```shell
+charmcraft fetch-lib charms.mutual_tls_interface.v0.mutual_tls
+```
+
+Add the following libraries to the charm's `requirements.txt` file:
+- jsonschema
+
+### Provider charm
+The provider charm is the charm providing public certificates to another charm that requires them.
+
+Example:
+```python
+from ops.charm import CharmBase, RelationJoinedEvent
+from ops.main import main
+
+from lib.charms.mutual_tls_interface.v0.mutual_tls import MutualTLSProvides
+
+
+class DummyMutualTLSProviderCharm(CharmBase):
```

Comment:
All the dummies should be gone with https://github.com/canonical/mutual-tls-interface/pull/3
:D